components/filters/FilterPicker.tsx

```diff
@@ -0,0 +1,187 @@
+import { DataList } from "@/components/DataList";
+import { Filter } from "@/lib/api";
+import { useWindowWidth } from "@/lib/hooks";
+import { useLayoutEffect, useRef, useState } from "react";
+import theme from "styles/theme";
+import Selectable, { SelectableSize } from "../form/Selectable";
+import FilterPickerCategory from "./FilterPickerCategory";
+
+export interface PickerFilter extends Filter {
+  active?: boolean;
+}
+
+interface FilterPickerProps {
+  filtersList: PickerFilter[];
+  activeFilters: PickerFilter[];
+  onFilterClick: (id?: string, filterType?: string) => any;
+  onFilterSeclect: (filterSelect?: string, enable?: boolean) => any;
+  selectedMemberCount?: number;
+}
+
+export default function FilterPicker({
+  filtersList,
+  activeFilters,
+  onFilterClick,
+  onFilterSeclect,
+  selectedMemberCount,
+}: FilterPickerProps) {
+  const width = useWindowWidth();
+  const [defaultHeight, setDefaultHeight] = useState<number>();
+  const [focusActive, setFocusActive] = useState<boolean>(true);
+  const [industryActive, setIndustryActive] = useState<boolean>();
+  const [regionActive, setRegionActive] = useState<boolean>();
+  const [experienceActive, setExperienceActive] = useState<boolean>();
+  const listRef = useRef<HTMLUListElement>();
+  const listItemsRef = useRef<HTMLLIElement[]>([]);
+  const filterIsSelected = activeFilters.length !== 0;
+
+  useLayoutEffect(() => {
```

Comment:
It looks like we removed the limit of two rows per category with the **View All** button. Thinking this `useLayoutEffect` no longer actually does anything now.

**before:**
<img width="50%" alt="image" src="https://user-images.githubusercontent.com/5141111/195274153-220dbf0d-594f-411a-8172-fe2dea9cfc8e.png">

**after:**
<img width="50%" alt="image" src="https://user-images.githubusercontent.com/5141111/195274582-fa880802-d4db-4505-a11c-5619402b1b3b.png">


The behavior was a bit awkward at times but it did limit this situation for smaller screens. Do you think it'd be worth re-adding? Happy to try advising here if you were finding it unwieldy.

Reply:
@taylorkmho yea this was interesting and something we both talked about during my initial proposals for this. I definitely had that scenario a couple times where I'm like "wait am I seeing all the filtering options? oh right there's a _View All_ option I need to press". But then again I totally see the benefit of being able to hide a lot of it for smaller screens while also showing new users we have this capability.

I tried thinking about this and had some ideas like maybe we have that same type of "limit to two rows" on initial page load, but instead of a _View All_ option, the Focus category would oscilate colors between `theme.color.brand.faded` and `theme.color.brand.base` and the 2 rowed filter options would be slightly faded as well. Then when a user clicked on it, it would open up the rest of the options and act as normal. But this implementation was a bit complex in my head and didn't want to go too deep into the implementation without talking it over.

**TLDR**: I thought about it, had some complex ideas but ultimately thought it would be better to let you advise.

Reply:
If I were to throw the UX book at you, I'd lean on limiting the number of selections to something reasonable with an obvious control to increase that number. Folks have trouble keeping track of too many options.

But I'd hate for you to have to re-engineer what was built the first time. I imagine enough has changed that it's no longer 1:1 to re-add it.